### PR TITLE
React 16.8.x - The Captains Hooks

### PR DESCRIPTION
**Pull request by DanielMSchmidt**
## Testing

There should be no suprises, but as React affects everything, please click a bit through the UI :) 

## Trade-offs

- Added some linting for the hooks, following the [Rules of Hooks](https://reactjs.org/docs/hooks-rules.html)

## Screenshots

Hook(s)

### CODE REVIEW

**Comment by mperrotti** [comment]
I updated ui-kit's peerDependencies to include this version of React: https://github.com/dcos-labs/ui-kit/pull/293

**Comment by GeorgiSTodorov** [comment]
Build fails because of security issues.
```
WARN notice [SECURITY] lodash has the following vulnerability: 1 low. Go here for more details: https://nodesecurity.io/advisories?search=lodash&version=4.17.4 - Run `npm i npm@latest -g` to upgrade your npm version, and then `npm audit` to get more info.

WARN notice [SECURITY] deep-extend has the following vulnerability: 1 low. Go here for more details: https://nodesecurity.io/advisories?search=deep-extend&version=0.4.2 - Run `npm i npm@latest -g` to upgrade your npm version, and then `npm audit` to get more info.

WARN notice [SECURITY] macaddress has the following vulnerability: 1 critical. Go here for more details: https://nodesecurity.io/advisories?search=macaddress&version=0.2.8 - Run `npm i npm@latest -g` to upgrade your npm version, and then `npm audit` to get more info.

WARN notice [SECURITY] randomatic has the following vulnerability: 1 low. Go here for more details: https://nodesecurity.io/advisories?search=randomatic&version=1.1.7 - Run `npm i npm@latest -g` to upgrade your npm version, and then `npm audit` to get more info.
```

**Comment by DanielMSchmidt** [comment]
@GeorgiSTodorov The problem is an integrity check so, we just need to npm install again and check in the new lockfile

**Comment by nLight** [comment]
We should update to 16.8.1

**Comment by DanielMSchmidt** [comment]
@nLight @GeorgiSTodorov done :)

File: package.json [hunk 1]
```diff
@@ -65,10 +65,10 @@
     "prettycron": "0.10.0",
     "prop-types": "15.6.0",
     "query-string": "4.1.0",
-    "react": "16.7.0",
+    "react": "^16.8.1",
```
Comment (GeorgiSTodorov) [inline package.json]: Is the `^` intended?

Comment (DanielMSchmidt) [reply package.json]: AAAAAAH

**Comment by DanielMSchmidt** [comment]
@GeorgiSTodorov If you have time and energy feel free to fix it 😇

**Comment by GeorgiSTodorov** [comment]
@DanielMSchmidt , I will look into it when I finish https://github.com/dcos/dcos-ui/pull/3583 and other sprint tasks and reviews.

**Comment by DanielMSchmidt** [comment]
👍

**Comment by GeorgiSTodorov** [comment]
Build fails with `Dependency length doesn't match.`.

**Pull request merged**